### ISSUE TITLE
helm rollback unsuccessful with service account

### ISSUE THREAD

**Issue by mnsudhan** [helm/helm#6036]
**Description:**
We had a helm chart installed which didn't have service account created.
We later did a helm upgrade of the chart with service account included.
When we did a helm rollback of the chart to original revision (which didn't have service account), pods are in crashLoopBackoff trying to mount the service account token which was deleted during rollback.

**Steps to reproduce:**
1. Install helm chart version 0.1.0 from attachment. This chart version doesn't have service account. Verify that pods are in running state.
2. Perform helm upgrade with chart version 0.1.4 from attachment. This chart version brings in service account "hello-world-sa". Verify that pods are in running state.
3. Perform a helm rollback to previous version. 
After step 3, the new pod created goes to crashLoopBackoff state with error "**FailedMount  111s (x70 over 128m)  kubelet, minikube  MountVolume.SetUp failed for volume "hello-world-sa-token-htdxc" : secret "hello-world-sa-token-xxxxx" not found**
"

[helm_rollback_issue_with_sa.zip](https://github.com/helm/helm/files/3402685/helm_rollback_issue_with_sa.zip)

**Output of `helm version`:**
[root@e2ecr2 hello-world]# helm version
Client: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}

**Output of `kubectl version`:**
[root@e2ecr2 hello-world]# kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Minikube

**Comment by mnsudhan** [comment]
Are we doing something wrong in our chart due to which helm is still keeping reference to service account in deployment even after rollback? Any help would be appreciated.

**Comment by bacongobbler** [comment]
Seems to be a result of https://github.com/helm/helm/issues/7159. Closing in favour of that ticket. Thanks.

**Comment by 15989206052** [comment]
> Are we doing something wrong in our chart due to which helm is still keeping reference to service account in deployment even after rollback? Any help would be appreciated.

hey,I have the same isuse, do  #7159 resolve our issue?
